lib/memery.rb

```diff
@@ -68,19 +72,20 @@ def fresh?(ttl)
         end
       end
 
-      def define_memoized_method!(klass, method_name, condition: nil, ttl: nil)
-        method_key = "#{method_name}_#{object_id}"
+      attr_accessor :use_hashed_arguments
 
+      def define_memoized_method!(klass, method_name, condition: nil, ttl: nil)
         original_visibility = method_visibility(klass, method_name)
-        original_arity = klass.instance_method(method_name).arity
+        use_hashed_arguments = self.use_hashed_arguments
 
         define_method(method_name) do |*args, &block|
           if block || (condition && !instance_exec(&condition))
             return super(*args, &block)
           end
 
           cache_store = (@_memery_memoized_values ||= {})
-          cache_key = original_arity.zero? ? method_key : [method_key, *args].hash
+          cache_key = args.empty? ? method_name : [method_name, *args]
```

Comment:
I think in the case it would be better to have nested `if` like this:

```ruby
if args.empty?
  method_name
else
  cache_key = [method_name, *args]
  use_hashed_arguments ? cache_key.hash : cache_key
end
```

This way the code is clearer and also we will be forced to cover the case when `use_hashed_arguments = false` since there will be a separate branch for that case.